### DISCUSSION TITLE
Schedule backups on QNAP NAS

### DISCUSSION THREAD

**Discussion by bastelbert** [teslamate-org/teslamate#2074] [Q&A] [answered]
Hi,

I'm running Teslamate on a QNAP NAS and I would like to backup the database every few hours.
I wrote the following script which works perfectly when started manually in the shell as admin user:

`#/bin/sh
NOW=$(date +"%Y%m%d_%H%M%S")
FILE="/share/CACHEDEV5_DATA/BackupVol1/BAK_Container/teslamate/"$NOW"_teslamate.bck"
cd /share/CACHEDEV1_DATA/Container/container-station-data/application/teslamate
docker-compose exec -T database pg_dump -U teslamate teslamate > $FILE
echo "Neues Backup: $FILE"
echo "Die letzten Sicherungen:"
ls -ltr /share/CACHEDEV5_DATA/BackupVol1/BAK_Container/teslamate/*.bck`

Unfortunately, scheduled runs via crontab of the admin user won't work as expected. The script seems to run, but the created backup files are empty.

`0 6,14,22 * * * /share/CACHEDEV1_DATA/Container/container-station-data/application/teslamate/teslamatebackup.sh > /share/CACHEDEV1_DATA/Container/container-station-data/application/teslamate/lastbackup.log`

Has anyone a clue what's wrong here? Thank you!

**Comment by gryghst72** [comment, marked as answer]
Add the full path to docker-compose executable. That’s what I had to do.

**Reply by bastelbert** [reply]
Awesome, thank you! Now it works as expected.
